src/modules/Dashboard/features/Applet/Schedule/ImportSchedulePopup/ImportSchedulePopup.utils.tsx

```diff
@@ -0,0 +1,281 @@
+        </>
+      );
+    case ImportScheduleErrors.NotificationTime:
+      return (
+        <>
+          <Trans i18nKey={`importScheduleErrors[${ImportScheduleErrors.NotificationTime}]`}>
+            <strong>Notification Time</strong>. Valid data format: <strong>HH:mm or -</strong>.
+          </Trans>{' '}
+          {t('importScheduleErrors.updateReupload')}
+        </>
+      );
+    case ImportScheduleErrors.Frequency:
+      return (
+        <>
+          <Trans i18nKey={`importScheduleErrors[${ImportScheduleErrors.Frequency}]`}>
+            <strong>Frequency</strong>. Valid data format:{' '}
+            <strong>Always, Once, Daily, Weekly, Monthly, Weekdays</strong>.
+          </Trans>{' '}
+          {t('importScheduleErrors.updateReupload')}
+        </>
+      );
+    case ImportScheduleErrors.Date:
+      return (
+        <>
+          <Trans i18nKey={`importScheduleErrors[${ImportScheduleErrors.Date}]`}>
+            <strong>Date</strong>. Valid data format: <strong>dd MMM yyyy</strong>.
+          </Trans>{' '}
+          {t('importScheduleErrors.updateReupload')}
+        </>
+      );
+    case ImportScheduleErrors.StartEndTime:
+      return (
+        <>
+          <Trans i18nKey={`importScheduleErrors[${ImportScheduleErrors.StartEndTime}]`}>
+            <strong>Activity End Time</strong> should be greater than{' '}
+            <strong>Activity Start Time</strong>. Please update and reupload the file.
+          </Trans>{' '}
+          {t('importScheduleErrors.updateReupload')}
+        </>
+      );
+    case ImportScheduleErrors.BetweenStartEndTime:
+      return (
+        <>
+          <Trans i18nKey={`importScheduleErrors[${ImportScheduleErrors.BetweenStartEndTime}]`}>
+            <strong>Notification Time</strong> should be between{' '}
+            <strong>Activity Start Time</strong> and <strong>Activity End Time</strong>. Please
+            update and reupload the file.
+          </Trans>{' '}
+          {t('importScheduleErrors.updateReupload')}
+        </>
+      );
+    default:
+      return null;
+  }
+};
+
+const getActivitiesFieldsToCheck = (data: ScheduleExportCsv, isUploadedSchedule: boolean) =>
+  data.reduce(
+    (
+      acc: {
+        activityNames: string[];
+        hasInvalidStartTimeField: boolean;
+        hasInvalidEndTimeField: boolean;
+        hasInvalidNotification: boolean;
+        hasInvalidFrequency: boolean;
+        hasInvalidDate: boolean;
+        hasInvalidStartEndTime: boolean;
+        hasInvalidNotificationTime: boolean;
+      },
+      { activityName, frequency, startTime, endTime, notificationTime, date },
+    ) => {
+      acc.activityNames.push(activityName);
+
+      if (isUploadedSchedule) {
+        if (
+          (frequency === 'Always' && startTime !== '-') ||
+          (frequency !== 'Always' && !timeValidationRegex.test(startTime))
+        ) {
+          acc.hasInvalidStartTimeField = true;
+        }
```

Comment:
Is not possible to set error here getInvalidError(ImportScheduleErrors.StartTime)? to avoid checking flags in the hook?

Reply:
done